### ISSUE TITLE
rabbitmq warning reports connection_closed_abruptly due to celery ping heartbeat

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/framework#558]
ovs version:

```
"versions": {
            "openvstorage-backend-core": "1.7.0-rev.588.2e534bc-1",
            "alba": "0.9.5",
            "openvstorage-webapps": "2.7.0-rev.3451.9fa0267-1",
            "openvstorage-backend-webapps": "1.7.0-rev.588.2e534bc-1",
            "openvstorage-core": "2.7.0-rev.3451.9fa0267-1",
            "volumedriver-server": "5.7.2-dev.201604211506.042c8a8",
            "openvstorage": "2.7.0-rev.3451.9fa0267-1",
            "openvstorage-cinder-plugin": "1.2.0-rev.26.fd48d23-1",
            "openvstorage-backend": "1.7.0-rev.588.2e534bc-1",
            "volumedriver-base": "5.7.2-dev.201604211506.042c8a8",
            "arakoon": "1.9.3"
}
```

Every minute we received a warning report from rabbitmq:

```
=WARNING REPORT==== 30-May-2016::14:35:07 ===
closing AMQP connection <0.21929.600> (10.100.199.1:55464 -> 10.100.199.1:5672):
connection_closed_abruptly

=WARNING REPORT==== 30-May-2016::14:36:07 ===
closing AMQP connection <0.23050.600> (10.100.199.1:56894 -> 10.100.199.1:5672):
connection_closed_abruptly

=WARNING REPORT==== 30-May-2016::14:37:06 ===
closing AMQP connection <0.24076.600> (10.100.199.1:58346 -> 10.100.199.1:5672):
connection_closed_abruptly
```

the celery inspect ping is the problem (connection to rabbitmq is not properly closed):

```
root      6309  0.0  0.0   4448   780 ?        S    14:02   0:00 /bin/sh -c /usr/bin/celery inspect ping -b amqp://ovs:0penv5tor4ge@10.100.199.1// --timeout=5 2> /dev/null | grep OK | perl -pe 's/?\[[0-9;]*m//g' || true
root      6310  8.4  0.3  92288 29480 ?        S    14:02   0:00 python2.7 /usr/bin/celery inspect ping -b amqp://ovs:0penv5tor4ge@10.100.199.1// --timeout=5
```

[Celery ping](https://github.com/openvstorage/framework/blob/master/ovs/extensions/generic/heartbeat.py#L54)

Solution:

```
#!/usr/bin/python
from pprint import pprint
from celery import Celery

cel = Celery()

pprint(cel.control.inspect().ping())
cel.close()
```

output:

```
{u'celery@cmp01': {u'ok': u'pong'},
 u'celery@cmp02': {u'ok': u'pong'},
 u'celery@cmp03': {u'ok': u'pong'},
 u'celery@cmp04': {u'ok': u'pong'}}
```

**Comment by wimpers** [comment]
@jeroenmaelbrancke can you create a PR for this one?

**Comment by wimpers** [comment]
@jeroenmaelbrancke  does https://github.com/openvstorage/framework/commit/dccecbc76f17bf7187e16d680bf6bc9768e7a62f fix this issue? If yes, please put the ticket to closed.

**Comment by jeroenmaelbrancke** [comment]
Investigation, testing, approaching 2 different methods, create PR (2d)
@Wim celery inspect ping doesn't allow multiple brokers in the connection string.
If the first node is down the celery ping gets a connection refused error.
Fixed this by creating a task on every worker to update his own timestamp.

**Comment by khenderick** [comment]
Fixed by #565
